File: Java/easy/Exercism/resistor-color-duo/src/main/java/ResistorColorDuo.java
class ResistorColorDuo {
    int value(String[] colors) {
        String[] COLORS = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
        int resistanceValue = 0;

        for (int i = 0; i < 2; i++) {
            if(i == 0) {
                resistanceValue += java.util.Arrays.asList(COLORS).indexOf(colors[i]) * 10;
            } else {
                resistanceValue += java.util.Arrays.asList(COLORS).indexOf(colors[i]);
            }
        }

        return resistanceValue;
    }
}

File: Javascript/easy/Exercism/resistor-color-duo/resistor-color-duo.js
export const decodedValue = (encoded) => {
  let resistanceValue = 0;
  resistanceValue += colorCode(encoded[0]) * 10; 
  resistanceValue += colorCode(encoded[1]);

  return resistanceValue;
};

const colorCode = (color) => {
  const COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];
  return COLORS.indexOf(color);
};
